File: problem-solving/algorithms/warmup/plus-minus/plus-minus.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    static void plusMinus(int[] arr) {
        int n = arr.length;
        int positives = (int) Arrays.stream(arr).filter(x -> x > 0).count();
        int negatives = (int) Arrays.stream(arr).filter(x -> x < 0).count();
        int zeroes = (int) Arrays.stream(arr).filter(x -> x == 0).count();
        
        float positiveFraction = (float)positives / (float)n;
        float negativeFraction = (float)negatives / (float)n;
        float zeroFraction = (float)zeroes / (float)n;
        
        System.out.println(positiveFraction);
        System.out.println(negativeFraction);
        System.out.println(zeroFraction);
    }

    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int n = scanner.nextInt();
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        int[] arr = new int[n];

        String[] arrItems = scanner.nextLine().split(" ");
        scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

        for (int i = 0; i < n; i++) {
            int arrItem = Integer.parseInt(arrItems[i]);
            arr[i] = arrItem;
        }

        plusMinus(arr);

        scanner.close();
    }
}

File: problem-solving/algorithms/warmup/plus-minus/plus-minus.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {

    static void plusMinus(int[] arr) {
        int n = arr.Length;
        int positives = arr.Where(x => x > 0).Count();
        int negatives = arr.Where(x => x < 0).Count();
        int zeroes = arr.Where(x => x == 0).Count();
        
        float positiveFraction = (float)positives / (float)n;
        float negativeFraction = (float)negatives / (float)n;
        float zeroFraction = (float)zeroes / (float)n;
        
        Console.WriteLine(positiveFraction);
        Console.WriteLine(negativeFraction);
        Console.WriteLine(zeroFraction);
    }

    static void Main(string[] args) {
        int n = Convert.ToInt32(Console.ReadLine());

        int[] arr = Array.ConvertAll(Console.ReadLine().Split(' '), arrTemp => Convert.ToInt32(arrTemp));
        plusMinus(arr);
    }
}